minject/inject.py

```diff
@@ -91,6 +118,24 @@ def _is_type(key: "RegistryKey[T]") -> TypeGuard[Type[T]]:
     return isinstance(key, type)
 
 
+def _is_key_async(key: RegistryKey) -> bool:
```

Comment:
(not sure) It looks like `RegistryKey` is a parameterized type — does it need a type parameter here, or do Python type-checkers implicitly assume unconstrained parameters?

Reply:
Good eye! I am not sure either. I ran a little test and the following code snippet passes `mypy <filename>` but fails `mypy --disallow-any-generics <filename>` with a `error: Missing type parameters for generic type "GenericType"` error:

```python
from typing import Generic, TypeVar

T = TypeVar('T')

class GenericType(Generic[T]):
    def __init__(self, value: T) -> None:
        self.value = value

    def get_value(self) -> T:
        return self.value

def test_typing(g : GenericType) -> bool:
    return bool(g.get_value())
```

Adding a param to `GenericType` in `test_typing` passes type checking.

My guess is that by default Generic types default to specifying an `Any` parameter based on the naming of the [`disallow_any_generics`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-disallow_any_generics) option.

I'll update the code to explicitly pass `Any`.